mne/evoked.py

```diff
@@ -881,14 +881,24 @@ def combine_evoked(all_evoked, weights):
     -----
     .. versionadded:: 0.9.0
     """
+    naves = np.array([evk.nave for evk in all_evoked], float)
     if isinstance(weights, str):
         _check_option('weights', weights, ['nave', 'equal'])
         if weights == 'nave':
-            weights = np.array([e.nave for e in all_evoked], float)
-            weights /= weights.sum()
-        else:  # == 'equal'
-            weights = [1. / len(all_evoked)] * len(all_evoked)
-    weights = np.array(weights, float)
+            weights = naves / naves.sum()
+            new_nave = np.sum(naves)
+        else:
+            weights = np.ones_like(naves)
+            new_nave = 1. / np.sum(1. / naves)
+        # for the new_nave calculation, cf. Matti's manual (pp 128-129); both
+        # of the formulae above are equivalent to the general formula below,
+        # but reduce numerical error.
+    else:
+        weights = np.array(weights, float)
+        new_nave = 1. / np.sum(weights ** 2 / naves)
```

Comment:
could you unify these `new_nave` calculations into one formula? or is that not possible?

Reply:
If I use the unified formula, then some of the tests in [test_arithmetic](https://github.com/mne-tools/mne-python/blob/master/mne/tests/test_evoked.py#L461) will fail, because `nave` will end up as, e.g., `30.00000001` instead of `30`.

Reply:
I'd still rather use the unified formula, it makes it clear the computation is the same in all cases, and that's not clear in the code any more. I would change the tests to use `allclose`

Reply:
OK, I've done this, though I'm fully expecting mailing list complaints saying "why is my `nave` 30.00001 now?"  For the record I'd much rather use 3 different (equivalent) formulae that avoid numerical error when possible, and leave detailed comments in the code showing why they're equivalent.

Reply:
why not use: https://docs.scipy.org/doc/numpy/reference/generated/numpy.round_.html for that then?

Reply:
@jasmainak because we don't *always* want to round, and if you're going to do something special for the cases where numerical error arises, why not do something special that avoids the error in the first place?